test/zero_one/geni/dataset_creation_test.clj

```diff
@@ -170,7 +170,23 @@
       (instance? Dataset dataset) => true
       (g/collect-vals dataset) => [[0 "A" false]
                                    [1 "B" false]
-                                   [2 "C" false]])))
+                                   [2 "C" false]]))
+  (fact "should work for map columns"
+    (let [dataset (g/records->dataset
+                   @tr/spark
+                   [{:i 0 :s "A" :b {:z ["a" "b"]}}
+                    {:i 1 :s "B" :b {:z ["c" "d"]}}])]
+      (instance? Dataset dataset) => true
+      (g/collect-vals dataset) => [[0 "A" {:z ["a" "b"]}]
+                                   [1 "B" {:z ["c" "d"]}]]))
+  (fact "should work for list of map columns"
+    (let [dataset (g/records->dataset
+                   @tr/spark
+                   [{:i 0 :s "A" :b [{:z 1} {:z 2}]}
+                    {:i 1 :s "B" :b [{:z 3}]}])]
+      (instance? Dataset dataset) => true
+      (g/collect-vals dataset) => [[0 "A" [{:z 1} {:z 2}]]
+                                   [1 "B" [{:z 3}]]])))
```

Comment:
@WaqasAliAbbasi what happens if there are missing keys to some of the maps in the list? Does it work as expected?

Reply:
Nice catch, it doesn't 😬
```clojure
(-> (g/records->dataset
     @tr/spark
     [{:i 0 :s "A" :b [{:z 1 :y true} {:z 2}]}
      {:i 1 :s "B" :b [{:z 3}]}])
    g/collect)
; Execution error (ArrayIndexOutOfBoundsException) at org.apache.spark.sql.catalyst.expressions.GenericInternalRow/genericGet (rows.scala:201).
; Index 1 out of bounds for length 1
```
Lemme try fixing it

Reply:
Also, perhaps we can also try what happens if you give different types to the same key? So one `:z` can be `"abc"` and another can be `1`?

Reply:
Added handling of missing keys.

For different types spark just throws an error as before:
```clojure
(-> (g/records->dataset
     @tr/spark
     [{:i 0 :s "A" :b [{:z "abc" :y true} {:z 2}]}
      {:i 1 :s "B" :b [{:z 3}]}])
    g/collect)
; Execution error (IllegalArgumentException) at org.apache.spark.sql.catalyst.CatalystTypeConverters$StringConverter$/toCatalystImpl (CatalystTypeConverters.scala:297).
; The value (2) of the type (java.lang.Long) cannot be converted to the string type
```